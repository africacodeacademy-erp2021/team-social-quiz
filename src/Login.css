
  .designcolor {
    color: #61a0d2;
  }
  
  .all_sections {
    display: flex;
    width: 1590px;
    height: 900px;
    position: absolute;
  }
  
  .display_infor {
    width: 760px;
    height: 900px;
    background: #22285F;
   
     
  }
 
  
  .login {
    margin-top: 5px;
    background-image: url(blob.svg);
    background-position: center;
    background-repeat: no-repeat;
    height: 800px;
    width: 900px;
    left: 100px;
    top: 0px;
    border-radius: 0px;
  
  }
  
  .h1_title {
    color: white;
    margin-left: 35px;
    margin-top: -2px;
    font-family: Open Sans;
    font-style: normal;
    font-weight: bold;
    font-size: 36px;
    line-height: 49px;
  }
  
  .h2_title {
    color: white;
    margin-left: 35px;
    margin-top: -2px;
    font-family: Open Sans;
    font-style: normal;
    font-weight: bold;
    font-size: 36px;
    line-height: 49px;
  }
  .h3_title {
    color: white;
    margin-left: 35px;
    font-family: Open Sans;
    font-weight: 600;
    font-size: 24px;
    line-height: 33px;
  }
  .p_title {
    color: white;
    display: inline-block;
    margin: 3px;
    font-family: Open Sans;
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    line-height: 25px;
  }
  
  .nav_newAccount {
    margin-left: 35px;
    padding: 2px;
  }
  
  .buttons_container {
    margin-left: 65px;
  }
  
  .link_text {
    color: #61d7f1;
    text-decoration: none;
    font-family: Open Sans;
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    line-height: 25px;
  }
  
  .btn_google {
    border: 0px;
    width: 300px;
    height: 33px;
    text-align: left;
    border-radius: 20px;
    color: black;
    background-color: white;
  
    font-family: Open Sans;
    font-style: normal;
    font-weight: normal;
    font-size: 21px;
    line-height: 33px;
  }
  
  .btn_github {
    border: 0px;
    width: 300px;
    height: 33px;
    text-align: left;
    border-radius: 20px;
    color: rgb(20, 81, 173);
    background-color: black;
    color: white;
    font-family: Open Sans;
    font-style: normal;
    font-weight: normal;
    font-size: 21px;
    line-height: 33px;
  }
  
  .mylogo {
    margin-right: 2px;
    margin-left: 5px;
    border-radius: 50%;
    margin-top: 2px;
  }
  
  .btn_githubinfo {
    background-color: black;
    color: white;
    height: 34px;
    border: 0px;
    font-size: 21px;
    left: 21.59%;
    right: 22.37%;
    top: 13.73%;
    bottom: 21.57%;
    line-height: 33px;
    float: center;
    text-align: right;
  }
  
  .btn_googleinfo {
    height: 24px;
    border: 0px;
    font-size: 21px;
    left: 21.59%;
    right: 22.37%;
    top: 13.73%;
    bottom: 21.57%;
    line-height: 33px;
  }
  
  .btn_facebook {
    border: 0px;
    width: 300px;
    height: 33px;
    text-align: left;
    border-radius: 20px;
  
    background-color: #3c5a9a;
  
    font-family: Open Sans;
    font-style: normal;
    font-weight: normal;
    font-size: 21px;
    line-height: 33px;
  }
  
  .my-facebook-button-class {
    height: 34px;
    border: 0px;
    font-size: 21px;
    left: 21.59%;
    right: 22.37%;
    top: 13.73%;
    bottom: 21.57%;
    line-height: 33px;
    background-color: #3c5a9a;
    color: white;
  }
  
  .myfacebooklogo {
    margin-right: 2px;
    margin-left: 5px;
    border-radius: 50%;
    margin-top: 2px;
  }
  
  .vector{
    position: absolute;
    left: 130px;
    top: 149px;
    height: 662.256591796875px;
    width: 669.755859375px;
    border-radius: 0px;
    mix-blend-mode: passthrough;
    
  }
  
  .blob{
    mix-blend-mode: saturation;
  }
  