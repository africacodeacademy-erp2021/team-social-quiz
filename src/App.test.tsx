import { mount, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { configure } from "enzyme";

import "@testing-library/jest-dom";
import React from "react";
import Facebook from "./authentication-and-login/facebook_authentication/facebook";
import Github from "./authentication-and-login/github_authentication/github";
import Google from "./authentication-and-login/google_authentication/google";
import Login from "./authentication-and-login/login/login";
import Registration from "./authentication-and-login/registration/registration";
import { render, screen } from "@testing-library/react";



configure({ adapter: new Adapter() });

describe("Facebook Button", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Facebook />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Github Button", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Github />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Google Button", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Google />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Login Page", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Registration Page", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Registration />);
    expect(wrapper).toMatchSnapshot();
  });
});


  it('check if Authentication popup opens after clicking on Google button', () => {  
    const login = mount(<Google />),
        buttonclick = login.find("button");
       
    buttonclick.first().simulate('click');
    expect(login.find('button')).toHaveLength(1);
});

test('Authentication button exists', async () => {
    render(<Facebook />);
 
    const button = screen.getByRole('button', { name: 'Login with Facebook' });
    expect(button).toBeInTheDocument();

})

test('Props values are passed correctly', async () => {
    render(<Facebook buttonName="SignIn with Facebook"/>);
 
    const vanillaInput = screen.getByRole('button', { name: 'SignIn with Facebook' });
    expect(vanillaInput).toBeInTheDocument();
})






  
