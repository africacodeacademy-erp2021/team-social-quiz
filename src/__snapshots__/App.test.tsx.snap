// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Facebook Button renders correctly 1`] = `ShallowWrapper {}`;

exports[`Github Button renders correctly 1`] = `ShallowWrapper {}`;

exports[`Google Button renders correctly 1`] = `ShallowWrapper {}`;

exports[`Login Page renders correctly 1`] = `ShallowWrapper {}`;

exports[`Registration Page renders correctly 1`] = `ShallowWrapper {}`;

exports[`GameStart renders GameStart correctly 1`] = `ShallowWrapper {}`;

exports[`ProgressBar renders ProgressBar correctly 1`] = `ShallowWrapper {}`;

exports[`Timer renders Timer correctly 1`] = `ShallowWrapper {}`;
